doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1814 @@
+L<SSL_set0_tmp_dh_pkey(3)>. There is no direct replacement for the "callback"
+functions. The callback was originally useful in order to have different
+parameters for export and non-export ciphersuites. Export ciphersuites are no
+longer supported by OpenSSL. Use of the callback functions should be replaced
+by one of the other methods described above.
+
+=item SSL_CTX_set_tlsext_ticket_key_cb.
+
+Use the new L<SSL_CTX_set_tlsext_ticket_key_evp_cb(3)> function instead.
+
+=item WHIRLPOOL, WHIRLPOOL_Init, WHIRLPOOL_Update, WHIRLPOOL_Final and WHIRLPOOL_BitUpdate
+
+See L</Deprecated low-level digest functions>.
+The Whirlpool algorithm has been moved to the L<Legacy Provider|/Legacy Algorithms>.
+
+=item X509_certificate_type
+
+This was an undocumented function. Applications can use L<X509_get0_pubkey(3)>
+and L<X509_get0_signature(3)> instead.
+
+=item X509_http_nbio and X509_CRL_http_nbio
+
+Use L<X509_load_http(3)> and L<X509_CRL_load_http(3)> instead.
+
+=back
+
+=head2 Using the FIPS Module in applications
+
+See L<fips_module(7)> and L<OSSL_PROVIDER-FIPS(7)> for details.
+
+=head2 OpenSSL command line application changes
+
+=head3 New applications
+
+L<'kdf'|openssl-kdf(1)> uses the new L<EVP_KDF(3)> API.
+L<'mac'|openssl-mac(1)> uses the new L<EVP_MAC(3)> API.
+
+=head3 Added options
+
+'-provider_path' and '-provider' are available to all apps and can be used
+multiple times to load any providers, such as the 'legacy' provider or
+third party providers. If used then the 'default' provider would also need to be
+specified if required. The '-provider_path' must be specified before the
+'-provider' option. 
+
+'-providers' used by 'list', which will display the list of loaded providers,
+their names, version and status.  It optionally displays their gettable parameters.
+
+'-objects' used by 'list' which will display the list of built in objects,
+i.e. OIDs with names.
+
+`-crl_lastupdate` and `-crl_nextupdate` used by 'ca' allows explicit setting of
+fields in the generated CRL.
+
+=head3 Removed options
+
+Interactive mode is not longer available.
+
+The `-crypt` option used by `passwd`.
+The '-C' option used by `x509`, `dhparam`, `dsaparam`, and `ecparam`.
+
+=head3 Other Changes
+
+The output of Command line applications may have minor changes.
+These are primarily changes in capitalisation and white space.  However, in some
+cases, there are additional differences.
+For example, the DH parameters output from `dhparam` now lists 'P', 'Q', 'G' and
+'pcounter' instead of 'prime', 'generator', 'subgroup order' and 'counter'
+respectively.
+
+The openssl commands that read keys, certificates, and CRLs now
+automatically detect the PEM or DER format of the input files so it is not
+necessary to explicitly specify the input format anymore. However if the
+input format option is used the specified format will be required.
+
+`speed` no longer uses low-level API calls.
+This implies some of the performance numbers might not be fully comparable
```

Comment:
`d:fully:`  It's doing something very different to what it used to.  People will still compare the outputs of course.